addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -646,28 +646,30 @@ var SnippetEditor = Widget.extend({
         const proms = Object.values(this.styles).map(opt => {
             return opt.updateUIVisibility();
         });
-        await Promise.all(proms);
+        // Get information about the visibility of options (except the ones
+        // located in the overlay). This is needed to check if the editor has
+        // visible options outside the options section.
+        const visibleOptions = await Promise.all(proms).then((OptionsVisibilityValues) =>
+            OptionsVisibilityValues.some((OptionsVisibilityValue) => OptionsVisibilityValue)
+        );
         // Hide the snippetEditor if none of its options are visible
         // This cannot be done using the visibility of the options' UI
-        // because some options can be located in the overlay.
-        const $visibleOptions = this.$optionsSection.find('we-top-button-group, we-customizeblock-option')
-                .children(':not(.d-none)');
+        // because some options can be located in the overlay / toolbar.
+        const visibleOptionsInSection = this.$optionsSection.find('we-top-button-group, we-customizeblock-option')
+            .children(':not(.d-none)').length;
```

Comment:
> We still need to check the visibility of the section's content only because the updateUIVisibility check returns true even if an option is located outside the section (e.g. in the toolbar).

Well it should not, that's the whole point of doing it that way :wink: But yes, let's review that in master and continue hacky DOM-check ways here :+1: